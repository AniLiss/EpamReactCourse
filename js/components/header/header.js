import React from 'react';
import {Search} from '../search/search';
import {SearchButton} from '../search-btn/serch-btn';
import {SearchResults} from '../search-results/search-results';

export const Header = () => {
    return (
        <header className="header">
            <nav className="header__nav">
                <h1>netflixroulette</h1>
                <SearchButton />
            </nav>
        <Search />
        <SearchResults />
        </header>
    )
};
