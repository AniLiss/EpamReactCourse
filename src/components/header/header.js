import React from 'react';
import {Search} from '../search/search';
import {Button} from '../search-btn/serch-btn';
import {SearchResults} from '../search-results/search-results';
import {Movie} from '../movie-info/movie-info';
import {Logo} from '../logo/logo';

import './header.scss';

export const Header = () => {
    return (
        <header className="header">
            <nav className="header__nav">
                <Logo />
                <Button text="Search" light />
            </nav>
            <Search />

            <Movie />
            <SearchResults />
        </header>
    )
};
