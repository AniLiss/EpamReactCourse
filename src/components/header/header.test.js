import {mount} from 'enzyme';
import React from 'react';
import {Header} from './header';

describe('Header', () => {
    it('is expected to render header correct', () => {
        const wrapper = mount(<Header />);
        expect(wrapper).toMatchSnapshot();
    })
});
