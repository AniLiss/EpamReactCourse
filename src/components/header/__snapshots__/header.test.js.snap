// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is expected to render header correct 1`] = `
<Header>
  <header
    className="header"
  >
    <nav
      className="header__nav"
    >
      <Logo>
        <h1
          className="logo"
        >
          netflixroulette
        </h1>
      </Logo>
      <Button
        light={true}
        text="Search"
      >
        <button
          className="search-btn  search-btn--light"
        >
          Search
        </button>
      </Button>
    </nav>
    <Search>
      <div
        className="search"
      >
        <h2
          className="search__title"
        >
          Find your movie
        </h2>
        <input
          className="search__input"
        />
        <div
          className="search__filters"
        >
          <h2
            className="search__filters-title"
          >
            Search by
          </h2>
          <Button
            active={true}
            text="Title"
          >
            <button
              className="search-btn search-btn--active "
            >
              Title
            </button>
          </Button>
          <Button
            text="Genre"
          >
            <button
              className="search-btn  "
            >
              Genre
            </button>
          </Button>
        </div>
        <Button
          active={true}
          text="Search"
        >
          <button
            className="search-btn search-btn--active "
          >
            Search
          </button>
        </Button>
      </div>
    </Search>
    <Movie>
      <section
        className="movie"
      >
        <div
          className="movie__img"
        >
          <img
            alt="Pulp Fiction"
            height="400"
            width="200"
          />
        </div>
        <div
          className="movie__info"
        >
          <h2
            className="movie__title"
          >
            Pulp Fiction
          </h2>
          <h3
            className="movie__short-desc"
          >
            Oscar Winning Movies
          </h3>
          <span
            className="movie__year"
          >
            1994
          </span>
          <span
            className="movie__duration"
          >
            154 min
          </span>
          <p
            className="movie__long-desc"
          >
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
          </p>
        </div>
      </section>
    </Movie>
    <SearchResults>
      <div
        className="search-results"
      >
        <div
          className="search-results__movies"
        >
          7 movies found
        </div>
        <ul
          className="search-results__filters-list"
        >
          <li
            className="search-results__filters-item"
          >
            Sort by
          </li>
          <li
            className="search-results__filters-item"
          >
            release date
          </li>
          <li
            className="search-results__filters-item"
          >
            rating
          </li>
        </ul>
        <div
          className="search-results__suggestions"
        >
          Films by 
          <span>
            Drama genre
          </span>
        </div>
      </div>
    </SearchResults>
  </header>
</Header>
`;
